lib/chef/provider/package/chocolatey.rb

```diff
@@ -0,0 +1,220 @@
+#
+# Copyright:: Copyright (c) 2015 Chef Software, Inc.
+# License:: Apache License, Version 2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+require 'chef/provider/package'
+require 'chef/resource/chocolatey_package'
+require 'chef/mixin/powershell_out'
+
+class Chef
+  class Provider
+    class Package
+      class Chocolatey < Chef::Provider::Package
+        include Chef::Mixin::PowershellOut
+
+        provides :chocolatey_package, os: "windows"
```

Comment:
I *think* these days it doesn't really matter, though, and if someone just does one of these in a library file:

```
Chef::Resource::ChocolateyPackage.provides :package, platform: :windows
```

That it'll pick that up, and most-specific-and-last-writer-wins (after john's 12.4.x changes) so I think all the ProviderPriorityMap stuff is now deprecated-ish, or at least unnecessary to get the job done...